packages/pages-shared/src/asset-server/rulesEngine.ts

```diff
@@ -48,9 +48,28 @@ export const generateRulesMatcher = <T>(
 				rule = rule.split(host_match[0]).join(`(?<${host_match[1]}>[^/.]+)`);
 			}
 
-			const path_matches = rule.matchAll(PLACEHOLDER_REGEX);
-			for (const path_match of path_matches) {
-				rule = rule.split(path_match[0]).join(`(?<${path_match[1]}>[^/]+)`);
+			const split = rule.split(/\?(?!<splat>)/g);
```

Comment:
I think we'll need some tests for this. File is here: `packages/pages-shared/__tests__/asset-server/rulesEngine.test.ts`

Reply:
This logic was changed in [9985580](https://github.com/cloudflare/wrangler2/pull/1863/commits/9985580230dc4eed5f78b95cfe7eff2d3bfa8d5f), how should I proceed?